### PR TITLE
Handle additional poetry source type

### PR DESCRIPTION
**Pull request by louislang**
Adds an additional source type to the parser. 

Closes https://github.com/phylum-dev/cli/issues/682

## Checklist

- [x] Does this PR have an associated issue (i.e., `closes #<issueNum>` in description above)?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [x] Have you created sufficient tests?
- [x] Have you updated all affected documentation?

### CODE REVIEW

**Comment by peterjmorgan** [comment]
Please ping me when this PR is merged and the associated release is ready

**Review by maxrake — APPROVED** [review]
LGTM.
For reference, https://github.com/python-poetry/poetry/issues/2571 talks about [the introduction of the "legacy" package source](https://github.com/python-poetry/poetry/pull/1635/files) as valid due to consistent terminology used in the PyPA community.

File: cli/src/lockfiles/python.rs [hunk 1]
```diff
@@ -104,7 +104,9 @@ impl Parse for Poetry {
             .packages
             .drain(..)
             .filter(|package| {
-                package.source.as_ref().map_or(true, |source| source.source_type == "git")
+                package.source.as_ref().map_or(true, |source| {
+                    source.source_type == "git" || source.source_type == "legacy"
+                })
```
Comment (cd-work) [inline cli/src/lockfiles/python.rs:109]: We do actually support submitting `git` dependencies to the backend, right? It seems like that's incorrectly disabled right now?

We might want to open an issue for supporting git dependencies with Poetry, not sure if that's desired?

**Review by cd-work — APPROVED** [review]
Assuming that we cannot analyze these dependencies, this PR does seem to correctly ignore them.

**Pull request merged**